Please provide support for non-setuptools.extern bundled packages

**Describe the bug**
The package currently uses `packaging`, `importlib_metadata` and `tomli` bundled with setuptools via `setuptools.extern` package. Setuptools is unbundled on Gentoo, therefore `setuptools.extern` is not provided and cx_Freeze fails due to missing imports, e.g.:

```pytb
____________________________________________ ERROR collecting tests/test_winversioninfo.py ____________________________________________
ImportError while importing test module '/tmp/cx_Freeze/tests/test_winversioninfo.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_winversioninfo.py:5: in <module>
    from cx_Freeze.winversioninfo import Version, VersionInfo
cx_Freeze/winversioninfo.py:12: in <module>
    from setuptools.extern.packaging import version as packaging_version
E   ModuleNotFoundError: No module named 'setuptools.extern'
```

Could you please add support for falling back to the top-level `packaging`, `tomli` and `importlib_metadata`, please?

**To Reproduce**
```
git clone https://github.com/marcelotduarte/cx_Freeze
cd cx_Freeze
python -m venv .venv
. .venv/bin/activate
pip install -r requirements-dev.txt

# this is equivalent to Gentoo's unvendoring
pip install jaraco.text more-itertools nspektr ordered-set packaging platformdirs tomli importlib_metadata
rm -r .venv/lib/python3.*/site-packages/setuptools/{extern,_vendor}
find .venv/lib/python3.*/site-packages/setuptools/ -name '*.py' -exec sed -e 's:from \w*[.]\+extern ::' -e 's:\w*[.]\+extern[.]::' -i {} +

python -m pytest
```

**Expected behavior**
Tests passing ;-).

**Desktop (please complete the following information):**
 - Platform information (e.g. Ubuntu Linux 22.04): Gentoo Linux
 - OS architecture (e.g. amd64): amd64
 - cx_Freeze version [e.g. 6.11]: 6.14.8, 2642a2579f355fea2a732e152ca762bd5ad9c98c
 - Python version [e.g. 3.10]: 3.11.2

--- a/ci/requirements.py
+++ b/ci/requirements.py
@@ -6,9 +6,12 @@ import sys
 from pathlib import Path
 
 try:
-    import tomllib
+    from tomllib import load as toml_load
 except ImportError:
-    from setuptools.extern import tomli as tomllib
+    try:
+        from setuptools.extern.tomli import load as toml_load
+    except ImportError:
+        from tomli import load as toml_load
 
 
 def main():
@@ -18,7 +21,7 @@ def main():
         print("pyproject.toml not found", file=sys.stderr)
         sys.exit(1)
     with pyproject_toml.open("rb") as file:
-        config = tomllib.load(file)
+        config = toml_load(file)
 
     root_dir = pyproject_toml.parent
     requirements = root_dir / "requirements.txt"

--- a/cx_Freeze/_compat.py
+++ b/cx_Freeze/_compat.py
@@ -9,9 +9,18 @@ from pathlib import Path
 if sys.version_info >= (3, 10):
     import importlib.metadata as importlib_metadata
 else:
-    from setuptools.extern import importlib_metadata
+    try:
+        from setuptools.extern import importlib_metadata
+    except ImportError:
+        import importlib_metadata
 
-__all__ = ["cached_property", "importlib_metadata"]
+try:
+    # pylint: disable-next=ungrouped-imports
+    from setuptools.extern import packaging
+except ImportError:
+    import packaging
+
+__all__ = ["cached_property", "importlib_metadata", "packaging"]
 __all__ += ["PLATFORM", "IS_LINUX", "IS_MACOS", "IS_MINGW", "IS_WINDOWS"]
 __all__ += ["IS_CONDA"]
 

--- a/cx_Freeze/command/_bdist_msi.py
+++ b/cx_Freeze/command/_bdist_msi.py
@@ -31,9 +31,7 @@ from sysconfig import get_platform, get_python_version
 from setuptools import Command
 from setuptools.errors import OptionError
 
-# pylint: disable-next=no-name-in-module
-from setuptools.extern.packaging import version as packaging_version
-
+from .._compat import packaging
 from ._pydialog import PyDialog
 
 
@@ -222,7 +220,7 @@ class bdist_msi(Command):
         version = metadata.get_version()
         # ProductVersion must be strictly numeric
         # XXX need to deal with prerelease versions
-        base_version = packaging_version.Version(version).base_version
+        base_version = packaging.version.Version(version).base_version
         # Prefix ProductName with Python x.y, so that
         # it sorts together with the other Python packages
         # in Add-Remove-Programs (APR)

--- a/cx_Freeze/command/bdist_msi.py
+++ b/cx_Freeze/command/bdist_msi.py
@@ -12,9 +12,7 @@ from sysconfig import get_platform
 
 from setuptools.errors import OptionError
 
-# pylint: disable-next=no-name-in-module
-from setuptools.extern.packaging import version as packaging_version
-
+from .._compat import packaging
 from ._bdist_msi import bdist_msi
 from ._pydialog import PyDialog
 
@@ -1031,7 +1029,7 @@ class BdistMSI(bdist_msi):
         author = self.distribution.metadata.get_contact() or "UNKNOWN"
         version = self.target_version or self.distribution.get_version()
         # ProductVersion must be strictly numeric
-        base_version = packaging_version.Version(version).base_version
+        base_version = packaging.version.Version(version).base_version
 
         # msilib is reloaded in order to reset the "_directories" global member
         # in that module.  That member is used by msilib to prevent any two

--- a/cx_Freeze/winversioninfo.py
+++ b/cx_Freeze/winversioninfo.py
@@ -8,8 +8,7 @@ import os
 from pathlib import Path
 from struct import calcsize, pack
 
-# pylint: disable-next=no-name-in-module
-from setuptools.extern.packaging import version as packaging_version
+from ._compat import packaging
 
 try:
     from win32verstamp import stamp as version_stamp
@@ -48,7 +47,7 @@ else:
     CX_FREEZE_STAMP = "internal"
 
 
-class Version(packaging_version.Version):
+class Version(packaging.version.Version):
     """A valid PEP440 version."""
 
 

--- a/tests/test_winmsvcr.py
+++ b/tests/test_winmsvcr.py
@@ -2,9 +2,14 @@
 
 from __future__ import annotations
 
+import sys
+
+import pytest
+
 from cx_Freeze.winmsvcr import FILES
 
 
+@pytest.mark.skipif(sys.platform != "win32", reason="Windows tests")
 def test_files():
     """Test winmsvcr.FILES."""
     expected = (

--- a/tests/test_winversioninfo.py
+++ b/tests/test_winversioninfo.py
@@ -2,9 +2,14 @@
 
 from __future__ import annotations
 
+import sys
+
+import pytest
+
 from cx_Freeze.winversioninfo import Version, VersionInfo
 
 
+@pytest.mark.skipif(sys.platform != "win32", reason="Windows tests")
 class TestVersionInfo:
     """Test VersionInfo class."""
 

